python/ray/tune/tune.py

```diff
@@ -299,17 +305,17 @@ def run(run_or_experiment,
     while not runner.is_finished():
         runner.step()
         if time.time() - last_debug > DEBUG_PRINT_INTERVAL:
-            if verbose:
-                reporter.report(runner)
+            if verbose and progress_reporter.should_report(runner):
+                progress_reporter.report(runner)
             last_debug = time.time()
 
     try:
         runner.checkpoint(force=True)
     except Exception:
         logger.exception("Trial Runner checkpointing failed.")
 
-    if verbose:
-        reporter.report(runner)
+    if verbose and progress_reporter.should_report(runner):
+        progress_reporter.report(runner)
```

Comment:
Can we not expose the `runner` and only expose the trials or some subset of the information?

Reply:
The reason is because I think the runner has a lot of internal logic that isn't required/shouldn't be exposed to the standard user.

Reply:
makes sense; passing in `trials, *sys_info` now (latter is a list of debug strings)